common_defs.h

```diff
@@ -711,6 +711,8 @@ rbit32(u32 v)
 	return v;
 }
 #define rbit32 rbit32
+#elif defined(_MSC_VER) && defined(ARCH_ARM32)
```

Comment:
Isn't it necessary to check whether the target CPU actually has the rbit instruction, i.e. is ARMv6T2 or later?  See what is done for gcc and clang just above this.  Or does MSVC not support targeting CPUs earlier than that?

Reply:
well, rbit is a very old inst set. Not sure if you can even find devices without it. Modern windows+arm will have it for sure.

Reply:
Well, modern Windows also wouldn't be arm32, right?

It's probably best to just leave this out if there isn't an easy way to do it correctly.  `rbit32()` isn't very important for performance anyway.

Reply:
yes, I think we can close it, but it's ok to merge it. I tested with gotbolt that the asm is identical to gcc/clang asm for arm32.

Reply:
I'm just going to close this.  If the intrinsic becomes available for arm64, we can add that.

Reply:
FYI, there is `__rbitll` in arm64intr.h starting with [16.2 visual studio](https://developercommunity.visualstudio.com/t/ARM64-missing-RBIT-intrinsics/498995).